traitsui/qt4/list_str_editor.py

```diff
@@ -196,6 +196,9 @@ def init(self, parent):
     def dispose(self):
         """ Disposes of the contents of an editor.
         """
+        self.model.beginResetModel()
+        self.model.endResetModel()
+
```

Comment:
This one is a drive-by: After adding the reset in the other editors but without this change in `list_str_editor`, the test suite is already free of warnings. But this looks like the right to do. We probably just did not have enough coverage to trigger the same warning for `ListStrEditor`.